src/software/simulation/er_force_simulator.cpp

```diff
@@ -123,18 +83,130 @@ void ErForceSimulator::setBallState(const BallState& ball_state)
+        simulator_setup_command->clear_set_team_yellow();
+    }
+
+    auto simulator_control = std::make_shared<sslsim::SimulatorControl>();
+    auto command_simulator = std::make_unique<amun::CommandSimulator>();
+
+    // Add each robot to be added to the teleport robot repeated field
+    for (const auto& robot_state_with_id : robots)
+    {
+        auto teleport_robot             = std::make_unique<sslsim::TeleportRobot>();
+        gameController::BotId* robot_id = new gameController::BotId();
+        robot_id->set_id(robot_state_with_id.id);
+
+        if (side == gameController::Team::BLUE)
+        {
+            robot_id->set_team(gameController::Team::BLUE);
+        }
+        else
+        {
+            robot_id->set_team(gameController::Team::YELLOW);
+        }
+
+        teleport_robot->set_x(static_cast<float>(
+            robot_state_with_id.robot_state.position().x() * MILLIMETERS_PER_METER));
+        teleport_robot->set_y(static_cast<float>(
+            robot_state_with_id.robot_state.position().y() * MILLIMETERS_PER_METER));
+        teleport_robot->set_allocated_id(robot_id);
+        teleport_robot->set_present(true);
+
+        teleport_robot->set_orientation(static_cast<float>(
+            robot_state_with_id.robot_state.orientation().toRadians()));
+
+        teleport_robot->set_v_x(static_cast<float>(
+            robot_state_with_id.robot_state.velocity().x() * MILLIMETERS_PER_METER));
+        teleport_robot->set_v_y(static_cast<float>(
+            robot_state_with_id.robot_state.velocity().y() * MILLIMETERS_PER_METER));
+        teleport_robot->set_v_angular(static_cast<float>(
+            robot_state_with_id.robot_state.angularVelocity().toRadians()));
+
+        *(simulator_control->add_teleport_robot()) = *teleport_robot;
+    }
+
+    // Send message to simulator to teleport robots
+    *(command_simulator->mutable_ssl_control())     = *simulator_control;
+    *(simulator_setup_command->mutable_simulator()) = *command_simulator;
+    er_force_sim->handleSimulatorSetupCommand(simulator_setup_command);
+
+    for (const auto& robot_state_with_id : robots)
+    {
+        auto simulator_robot = std::make_shared<ErForceSimulatorRobot>(
+            RobotStateWithId{.id          = robot_state_with_id.id,
+                             .robot_state = robot_state_with_id.robot_state},
+            robot_constants, wheel_constants);
+
+        if (side == gameController::Team::BLUE)
+        {
+            if (robot_state_with_id.id < blue_simulator_robots.size())
+            {
+                blue_simulator_robots[robot_state_with_id.id] = simulator_robot;
+            }
+            else
+            {
+                blue_simulator_robots.emplace_back(simulator_robot);
+            }
+        }
+        else
+        {
+            if (robot_state_with_id.id < yellow_simulator_robots.size())
+            {
+                yellow_simulator_robots[robot_state_with_id.id] = simulator_robot;
+            }
```

Comment:
This isn't accurate, because the index doesn't always equal robot id. Sometimes robots are taken off the field and we may want to simulate that. Instead, just clear the vectors, i.e. `yellow_simulator_robots.clear()` and `blue_simulator_robots.clear()`